lnutil/msglib.go

```diff
@@ -1,47 +1,770 @@
 
-	MSGID_SELFPUSH = 0x80
-)
+//----------
+
+//message with no information, just shows a point is requested
+type PointReqMsg struct {
+	PeerIdx uint32
+}
+
+func NewPointReqMsg(peerid uint32) *PointReqMsg {
+	p := new(PointReqMsg)
+	p.PeerIdx = peerid
+	return p
+}
+
+func NewPointReqMsgFromBytes(peerid uint32) (*PointReqMsg, error) {
+	return NewPointReqMsg(peerid), nil // no way to have error
+}
+
+func (self PointReqMsg) Bytes() []byte {
+	var msg []byte
+	msg = append(msg, self.MsgType())
+	return msg
+}
+
+func (self PointReqMsg) Peer() uint32   { return self.PeerIdx }
+func (self PointReqMsg) MsgType() uint8 { return MSGID_POINTREQ }
+
+//message to be used for reply to point request
+type PointRespMsg struct {
+	PeerIdx    uint32
+	ChannelPub [33]byte
+	RefundPub  [33]byte
+	HAKDbase   [33]byte
+}
+
+func NewPointRespMsg(peerid uint32, chanpub [33]byte, refundpub [33]byte, HAKD [33]byte) *PointRespMsg {
+	pr := new(PointRespMsg)
+	pr.PeerIdx = peerid
+	pr.ChannelPub = chanpub
+	pr.RefundPub = refundpub
+	pr.HAKDbase = HAKD
+	return pr
+}
+
+func NewPointRespMsgFromBytes(b []byte, peerid uint32) (*PointRespMsg, error) {
+	pm := new(PointRespMsg)
+	b = b[1:] // get rid of messageType
+	pm.PeerIdx = peerid
+
+	if len(b) != 99 {
+		return nil, fmt.Errorf("PointRespHandler err: msg %d bytes, expect 99\n", len(b))
+	}
+
+	copy(pm.ChannelPub[:], b[:33])
+	copy(pm.RefundPub[:], b[33:36])
+	copy(pm.HAKDbase[:], b[36:])
+
+	return pm, nil
+}
+
+func (self PointRespMsg) Bytes() []byte {
+	var msg []byte
+	msg = append(msg, self.MsgType())
+	msg = append(msg, self.ChannelPub[:]...)
+	msg = append(msg, self.RefundPub[:]...)
+	msg = append(msg, self.HAKDbase[:]...)
+	return msg
+}
+
+func (self PointRespMsg) Peer() uint32   { return self.PeerIdx }
+func (self PointRespMsg) MsgType() uint8 { return MSGID_POINTRESP }
+
+//message with a channel's description
+type ChanDescMsg struct {
+	PeerIdx   uint32
+	Outpoint  wire.OutPoint
+	PubKey    [33]byte
+	RefundPub [33]byte
+	HAKDbase  [33]byte
+
+	Capacity    int64
+	InitPayment int64
+
+	ElkZero   [33]byte //consider changing into array in future
+	ElkOne    [33]byte
+	ElkTwo    [33]byte
+	Signature [64]byte
+}
+
+func NewChanDescMsg(peerid uint32, OP wire.OutPoint, pubkey [33]byte, refund [33]byte, hakd [33]byte,
+	capacity int64, payment int64, ELKZero [33]byte, ELKOne [33]byte, ELKTwo [33]byte) *ChanDescMsg {
+
+	cd := new(ChanDescMsg)
+	cd.PeerIdx = peerid
+	cd.Outpoint = OP
+	cd.PubKey = pubkey
+	cd.RefundPub = refund
+	cd.HAKDbase = hakd
+	cd.Capacity = capacity
+	cd.InitPayment = payment
+	cd.ElkZero = ELKZero
+	cd.ElkOne = ELKOne
+	cd.ElkTwo = ELKTwo
+	return cd
+}
+
+func NewChanDescMsgFromBytes(b []byte, peerid uint32) (*ChanDescMsg, error) {
+	cm := new(ChanDescMsg)
+	b = b[1:] // get rid of messageType
+	cm.PeerIdx = peerid
+
+	if len(b) != 250 {
+		return nil, fmt.Errorf("got %d byte channel description, expect 250", len(b))
+	}
+
+	var op [36]byte
```

Comment:
this works but can get annoying if you need to change things.  bytes.buffer can make this easier, because you can use buf.Next(33) to read out the later bytes.  Either way works though.